.github/workflows/e2e-tests.yml

```diff
@@ -146,6 +146,21 @@ jobs:
       - name: Build uberjar with ./bin/build.sh
         run: ./bin/build.sh
 
+      - name: Mark with the commit hash
+        run: echo ${{ github.sha }} > COMMIT-ID
+        shell: bash
+      - name: Calculate SHA256 checksum
+        run: sha256sum ./target/uberjar/metabase.jar > SHA256.sum
+        shell: bash
+      - name: Upload JARs as artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: metabase-${{ matrix.edition }}-${{ github.sha }}-uberjar
+          path: |
+            ./target/uberjar/metabase.jar
+            ./COMMIT-ID
+            ./SHA256.sum
+
```

Comment:
The only reason I didn't edit https://github.com/metabase/metabase/blob/master/.github/actions/prepare-uberjar-artifact/action.yml directly was that it's used in another workflow. I'd like to focus only on E2E for now to keep diffs as atomic as possible.

But the end goal will be using unified logic for both E2E and any other workflow that needs to build and store uberjar as an artifact.